tools/ippfind.c

```diff
@@ -336,18 +336,15 @@ main(int  argc,				/* I - Number of command-line args */
                                argv + i)) == NULL)
             return (IPPFIND_EXIT_MEMORY);
 
-          while (i < argc)
-            if (!strcmp(argv[i], ";"))
-              break;
-            else
-              i ++;
-
-          if (i >= argc)
+          do
           {
-            _cupsLangPrintf(stderr, _("ippfind: Expected semi-colon after %s."),
-                            "--exec");
-            show_usage();
-          }
+            if (strcmp(argv[i], ";"))
+            {
+              _cupsLangPrintf(stderr, _("ippfind: Expected semi-colon after %s."),
```

Comment:
Again, we are looking for ";" in argv. You need to loop through the argument list and stop when you see it. If you don't see it, you show an error.

Reply:
That's what this code does already?